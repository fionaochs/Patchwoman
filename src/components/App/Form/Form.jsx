import React from 'react';
import propTypes from 'prop-types';
// import styles from './Form.css';

const Form = ({ onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value="link" name="link" placeholder="API Route" onChange={onChange}></input>

    <input type="radio" value="get" name="method" onChange={onChange}></input>
    <label htmlFor="get">Get</label>

    <input type="radio" value="post" name="method" onChange={onChange}></input>
    <label htmlFor="post">Post</label>

    <input type="radio" value="put" name="method" onChange={onChange}></input>
    <label htmlFor="put">Put</label>

    <input type="radio" value="patch" name="method" onChange={onChange}></input>
    <label htmlFor="patch">Patch</label>

    <input type="radio" value="delete" name="method" onChange={onChange}></input>
    <label htmlFor="delete">Delete</label>

    <textarea placeholder="Raw JSON Body" name="jsonBody" onChange={onChange}></textarea>

    <button>Go</button>
  </form>
);

Form.propTypes = {
  onChange: propTypes.func.isRequired,
  onSubmit: propTypes.func.isRequired
};
export default Form;
