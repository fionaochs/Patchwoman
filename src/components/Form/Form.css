.tealButton {
  background-color: #140567;
  font-family: 'Open Sans', sans-serif;
  font-weight: bold;
  font-size: 1rem;
  color: white;
  border-radius: .3rem;
  border: none;
  height: 2.5rem;
  width: 7.3rem;
  margin-left: 3em;
  box-shadow: 3px 3px 5px 1px rgb(157, 170, 187);
  letter-spacing: 0.02rem;
}

.Form {  
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 45%;

  h2 {
    font-size: 1.5em;
    margin: 0;
  }

  div, section {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 100%;
  }

  input[type="text"] {
    width: 100%;
    max-width: 60rem;
    font-size: 0.8rem;
    padding-left: 1rem;
    height: 2.2rem;
    background-color: #d6eaf7;
  }

  textarea {
    height: 5rem;
    font-size: 0.8rem;
    padding: 1rem;
    background-color: #d6eaf7;
    font-weight: 400;
    width: 100%;
  }

  input[type="radio"] {
    display: none;
  }

  input[type="radio"]:checked + label {
    background-color: #d6eaf7;
    background-color: #d6eaf7;
    background-color: #d6eaf7;
    background-color: #d6eaf7;
    background-color: #d6eaf7;
}

  label:hover {
    background-color: #4238c4;
  }

  label {
    text-transform: uppercase;
    font-size: 1.2rem;
    color: white;
    line-height: 2.8em;
    height: 2.8em;
    vertical-align: middle;
    text-align: center;
    background-color: #140567;
    border-radius: .3rem;
    margin-right: 2.4rem;
    width: 8rem;
    min-width: 4.8rem;
  }

  label:last-of-type {
    margin-right: 0;
  }

  button:hover {
    background-color: #4238c4;
  }

  button, label {
    cursor: pointer;
  }
}