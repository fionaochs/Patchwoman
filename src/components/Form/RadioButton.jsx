import React from 'react';
import PropTypes from 'prop-types';

export const RadioButton = ({ name, value, onChange }) => {
  return (
    <>
      <input
        id={value}
        type="radio"
        name={name}
        value={value}
        onChange={onChange} />
      <label htmlFor={value}>{value}</label>
    </>
  );
};
RadioButton.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func,
  name: PropTypes.string
};
