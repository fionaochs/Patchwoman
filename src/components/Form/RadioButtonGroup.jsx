import React from 'react';
import PropTypes from 'prop-types';

export const RadioButtonGroup = ({ name, onChange, children }) => {
  const radioButtonsWithNameAndOnChange = React.Children.map(children, child => {
    return React.cloneElement(child, {
      name,
      onChange
    });
  });
  return (
    <>
      {radioButtonsWithNameAndOnChange}
    </>
  );
};
RadioButtonGroup.propTypes = {
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  children: PropTypes.array.isRequired
};
