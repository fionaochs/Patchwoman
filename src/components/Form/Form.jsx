import React from 'react';
import PropTypes from 'prop-types';
import { RadioButton } from './RadioButton.jsx';
import { RadioButtonGroup } from './RadioButtonGroup.jsx';
import styles from './Form.css';

const Form = ({
  onSubmit,
  onChange,
  url,
  body,
  buttonText = 'Submit'
}) => {
  return (
    <form className={styles.Form} onSubmit={onSubmit}>
      <div>
        <h2>Request</h2>
      </div>
      <div>
        <input
          type="text"
          name="url"
          value={url}
          onChange={onChange}
          placeholder="URL" />
        <button className={styles.button}>{buttonText}</button>
      </div>
      <section className={styles.buttons} >
        <RadioButtonGroup className={styles.buttons} name="method" onChange={onChange}>
          <RadioButton value="get" />
          <RadioButton value="post" />
          <RadioButton value="put" />
          <RadioButton value="patch" />
          <RadioButton value="delete" />
        </RadioButtonGroup>
      </section>
      <textarea
        type="text"
        name="body"
        value={body}
        onChange={onChange}
        placeholder="Raw JSON Body" />
    </form>
  );
};

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  body: PropTypes.string,
  url: PropTypes.string.isRequired,
  buttonText: PropTypes.string,
};

export default Form;
