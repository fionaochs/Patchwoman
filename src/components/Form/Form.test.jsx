import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.jsx';

describe('Form', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Form
        onSubmit={() => {}}
        onChange={() => {}}
        url="http://musicbrainz.org/ws/2/artist?query=nahko&fmt=json&limit=25"
        method="get"
        buttonText="Send" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
