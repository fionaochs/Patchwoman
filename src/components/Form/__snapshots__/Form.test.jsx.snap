// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches a snapshot 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <div>
    <h2>
      Request
    </h2>
  </div>
  <div>
    <input
      name="url"
      onChange={[Function]}
      placeholder="URL"
      type="text"
      value="http://musicbrainz.org/ws/2/artist?query=nahko&fmt=json&limit=25"
    />
    <button
      className="button"
    >
      Send
    </button>
  </div>
  <section
    className="buttons"
  >
    <RadioButtonGroup
      className="buttons"
      name="method"
      onChange={[Function]}
    >
      <RadioButton
        value="get"
      />
      <RadioButton
        value="post"
      />
      <RadioButton
        value="put"
      />
      <RadioButton
        value="patch"
      />
      <RadioButton
        value="delete"
      />
    </RadioButtonGroup>
  </section>
  <textarea
    name="body"
    onChange={[Function]}
    placeholder="Raw JSON Body"
    type="text"
  />
</form>
`;
